Content.Shared/Alert/AlertsSystem.cs

```diff
@@ -176,6 +187,31 @@ public override void Initialize()
         LoadPrototypes();
     }
 
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<AlertAutoRemoveComponent>();
+        while (query.MoveNext(out var uid, out var autoComp))
+        {
+            if (autoComp.Alerts.Count <= 0 || !TryComp<AlertsComponent>(uid, out var alertComp))
+            {
+                EntityManager.RemoveComponent<AlertAutoRemoveComponent>(uid);
+                continue;
+            }
+
+            foreach (var (alertKey,  alertState) in autoComp.Alerts)
+            {
+                if (alertState.AutoRemove is null || alertState.AutoRemove >= _timing.CurTime)
+                    continue;
+
+                autoComp.Alerts.Remove(alertKey);
+                alertComp.Alerts.Remove(alertKey);
+                Dirty(uid, alertComp);
+            }
+        }
+    }
```

Comment:
Subscribe to EntityUnpausedEvent for the component and bump all the timers by pausedtime (see existing implementations) so this still works if a server is saved and then loaded.

Also use RemCompDeferred and avoid touching the components during iteration.